### PR TITLE
[one-cmds] Add helper scripts for onnx_legalizer.py testing

### PR DESCRIPTION
**Pull request by binarman**
This PR adds onnx_legalize_run_compare.py and
print_onnx_model.py to test onnx_lagalizer correctness.

Related issue #8395

ONE-DCO-1.0-Signed-off-by: Alexander Efimov <a.efimov@samsung.com>

### CODE REVIEW

File: compiler/one-cmds/tests/onnx_legalize_run_compare.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            print("output {} shape mismatch".format(i))
+            return False
+        peak_error = np.abs(ref_outputs[i] - test_outputs[i]).max()/np.abs(ref_outputs[i]).max()
+        if peak_error > tolerance:
+            print("output {} peak error to value ratio {} is too big".format(i, peak_error))
+            return False
+    return True
+
+if __name__ == '__main__':
+    # this manipulation is needed to add searcg path where onnx_legalizer is installed
```
Comment (seanshpark) [inline compiler/one-cmds/tests/onnx_legalize_run_compare.py:38]: typo `searcg`

File: compiler/one-cmds/tests/onnx_legalize_run_compare.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    sys.path += os.environ['PATH'].split(':')
+    import onnx_legalizer
+
+
```
Comment (seanshpark) [inline compiler/one-cmds/tests/onnx_legalize_run_compare.py:42]: (does format checker require two lines after import? if not can we remove one?)

Comment (binarman) [reply compiler/one-cmds/tests/onnx_legalize_run_compare.py:42]: Oof, I just noticed that #8221 disabled checks for compiler/one-cmds directory: [these lines](https://github.com/Samsung/ONE/pull/8221/files#diff-f26ca7df49b7cc7c3f32f3769aa92113e722738d77a5f2458a5850dde86b352dR181-R182)

Comment (binarman) [reply compiler/one-cmds/tests/onnx_legalize_run_compare.py]: created issue for this: #8448

File: compiler/one-cmds/tests/onnx_legalize_run_compare.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    outputs = session.run(output_names, inputs)
+    return outputs
+
+
+def _compare_resuts(ref_outputs, test_outputs, tolerance):
+    num_outputs = len(ref_outputs)
+    assert (len(test_outputs) == num_outputs)
+    for i in range(num_outputs):
+        if ref_outputs[i].shape != test_outputs[i].shape:
+            print("output {} shape mismatch".format(i))
```
Comment (seanshpark) [inline compiler/one-cmds/tests/onnx_legalize_run_compare.py:32]: (optional) print `ref_outputs[i].shape` and `test_outputs[i].shape` itself may help debugging.

**Review by seanshpark — DISMISSED** [review]
LGTM

File: compiler/one-cmds/tests/onnx_legalize_run_compare.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import sys
+import os
+import numpy as np
+
+
+def _generate_inputs(model):
+    inputs = {}
+    for input in model.graph.input:
+        # check if elem type is float32
+        assert (input.type.tensor_type.elem_type == 1)
```
Comment (mhs4670go) [inline compiler/one-cmds/tests/onnx_legalize_run_compare.py:12]: I'm not sure it's impossible to use enum-like things instead of `1`?

File: compiler/one-cmds/tests/onnx_legalize_run_compare.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        input_shape = []
+        for dim in input.type.tensor_type.shape.dim:
+            input_shape += [dim.dim_value]
+        inputs[input.name] = np.random.random(input_shape).astype(np.float32)
+    return inputs
```
Comment (mhs4670go) [inline compiler/one-cmds/tests/onnx_legalize_run_compare.py:17]: (optional) It would be helpful to let readers know what input's spec is like with comment.

File: compiler/one-cmds/tests/onnx_legalize_run_compare.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+
+def _run_model(model, inputs):
+    output_names = list(map(lambda output: output.name, model.graph.output))
+    session = rt.InferenceSession(model.SerializeToString())
+    outputs = session.run(output_names, inputs)
+    return outputs
+
+
+def _compare_resuts(ref_outputs, test_outputs, tolerance):
```
Comment (mhs4670go) [inline compiler/one-cmds/tests/onnx_legalize_run_compare.py:27]: `resuts` is not type but intentional thing?

File: compiler/one-cmds/tests/onnx_legalize_run_compare.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        if peak_error > tolerance:
+            print("output {} peak error to value ratio {} is too big".format(
+                i, peak_error))
+            return False
+    return True
+
+
+if __name__ == '__main__':
+    # this manipulation is needed to add search path where onnx_legalizer is installed
+    sys.path += os.environ['PATH'].split(':')
```
Comment (mhs4670go) [inline compiler/one-cmds/tests/onnx_legalize_run_compare.py:45]: IIUC, this manipulation is possible because `runtestall.sh` adds `bin` directory to `PATH`. It would have better maintainability if the script receives `onnx_legalizer`'s path as an argument.

File: compiler/one-cmds/tests/onnx_legalize_run_compare.py [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    Args:
+        model (onnx.onnx_ml_pb2.ModelProto): target model
+
+    Returns:
+        dict from str to numpy.ndarray: generated inputs
+    """
+    inputs = {}
+    for input in model.graph.input:
+        # check if elem type is float32
+        # list of types could ne externded, this is a property of current testsuite
```
Comment (mhs4670go) [inline compiler/one-cmds/tests/onnx_legalize_run_compare.py:35]: This change might be another PR.
```suggestion
        # list of types could be externded, this is a property of current testsuite
```

**Review by mhs4670go — APPROVED** [review]
LGTM. Thank you:)
